src/heretic/schemas.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+
+@dataclass
+class GenerationStep:
+    step_index: int
+    token_id: int
+    token: str
+    logprob: float | None
+    topk: List[Dict[str, float]] | None = None
+    entropy: float | None = None
+
+    # Optional internals
+    last_hidden_state: List[float] | None = None
+    residuals_per_layer: List[List[float]] | None = None
+    attention_summary: Dict[str, Any] | None = None
+
+
+@dataclass
+class GenerationTrace:
+    steps: List[GenerationStep]
+    finish_reason: str | None = None
+
+
+@dataclass
+class ContextMetadata:
+    # model and generation params
+    system_prompt: str | None = None
+    model_name: str | None = None
+    generation_params: Dict[str, Any] | None = None
+
+
+@dataclass
+class ResponseMetadata:
+    # basic prompt stuff
+    prompt_text: str | None = None
+
+    # multi-turn in the future?
```

Comment:
Probably not. Let's leave such advanced stuff out for the moment. But it's good to think about those possibilities anyway!